### PR TITLE
Added test for version.go and resolved undefined error in that same file

### PR DESCRIPTION
**Pull request by Dhiraj240**
Need reviews on the tests.
Tests are on version file
#258 
#266

### CODE REVIEW

**Comment by andyzhangx** [comment]
@Dhiraj240 this test failed:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_azurefile-csi-driver/265/pull-azurefile-csi-driver-unit/1258204236047454213

**Comment by Dhiraj240** [comment]
Yes, it was due to changes I made to use structure in order to have efficiency but couldn't work. I will change it.

File: pkg/azurefile/version_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		t.Log("Got as expected")
+	} else {
+		t.Errorf("Unexpected Versions: Please check the requirements")
+	}
+	
+}
+
+func TestGetVersionYAML(t *testing.T) {
+	get := GetVersion()
+	response, err := yaml.Marshal(&get)
```
Comment (animeshk08) [inline pkg/azurefile/version_test.go:54]: Hi @Dhiraj240. I think your test is failing due to this line. Try changing response to _

**Comment by animeshk08** [comment]
Also, consider running `hack/update-gofmt.sh` before pushing the changes. It formats the Go source code :)

**Comment by Dhiraj240** [comment]
> Also, consider running `hack/update-gofmt.sh` before pushing the changes. It formats the Go source code :)

It says no  No such file or directory

**Comment by animeshk08** [comment]
This is the file: https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/hack/update-gofmt.sh

**Comment by Dhiraj240** [comment]
> > It says no No such file or directory
> 
> This is the file: https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/hack/update-gofmt.sh

I got this update-gofmt.sh: 17: set: Illegal option -o pipefail

**Comment by Dhiraj240** [comment]
Moreover I checked it using below and got error
bash -x  update-gofmt.sh
+ set -euo pipefail
+ find . -name '*.go'
+ grep -v '\/vendor\/'
+ xargs gofmt -s -w
error: cannot use -w with standard input

**Comment by Dhiraj240** [comment]
@animeshk08 Can you share your slack contact or any other?

File: pkg/azurefile/version_test.go (hunk 1 above)
Comment (Dhiraj240) [reply pkg/azurefile/version_test.go]: Oh, sorry.

**Comment by Dhiraj240** [comment]
@andyzhangx Test slightly increased coverage but stands done as of now.

File: pkg/azurefile/version_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+package azurefile
+
+import (
+	"testing"
+
+	"sigs.k8s.io/yaml"
+)
+
+func TestGetVersion(t *testing.T) {
+	get := VersionInfo{
```
Comment (andyzhangx) [inline pkg/azurefile/version_test.go:26]: You are not testing `GetVersion` func

File: pkg/azurefile/version_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	if get == want {
+		t.Log("Got as expected")
+	} else {
+		t.Errorf("Unexpected Versions: Please check the requirements")
+	}
+
+}
+
+func TestGetVersionYAML(t *testing.T) {
+	get := GetVersion()
```
Comment (andyzhangx) [inline pkg/azurefile/version_test.go:53]: Should test against `GetVersionYAML` func

Comment (Dhiraj240) [reply pkg/azurefile/version_test.go]: Changes done

File: pkg/azurefile/version_test.go (hunk 2 above)
Comment (Dhiraj240) [reply pkg/azurefile/version_test.go]: Tested now, without structure approach

File: pkg/azurefile/version_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	if get == want {
+		t.Log("Got as expected")
+	} else {
+		t.Errorf("Unexpected Versions: Please check the requirements")
+	}
+
+}
+
+func TestGetVersionYAML(t *testing.T) {
+	_, get := GetVersionYAML()
```
Comment (andyzhangx) [inline pkg/azurefile/version_test.go:48]: it's actually should be like this:
```
yaml, err := GetVersionYAML()
```
and you should compare the `yaml` with expected string and also check there is no `err`

File: pkg/azurefile/version_test.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	"runtime"
+	"testing"
+
+	"sigs.k8s.io/yaml"
+)
+
+func TestGetVersion(t *testing.T) {
+	get := GetVersion()
+
+	want := VersionInfo{
```
Comment (andyzhangx) [inline pkg/azurefile/version_test.go:30]: `want` --> `expected`

File: pkg/azurefile/version_test.go (hunk 4 above)
Comment (Dhiraj240) [reply pkg/azurefile/version_test.go:48]: alright

**Pull request closed without merging**